packages/@aws-cdk/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -39,12 +49,364 @@ export class RotationSchedule extends Resource {
+   */
+  public static postgreSqlSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.POSTGRESQL_SINGLE_USER,
+    });
+  }
+
+  /**
+   * PostgreSQL Multi User
+   */
+  public static postgreSqlMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.POSTGRESQL_MULTI_USER,
+    });
+  }
+
+  /**
+   * Oracle Single User
+   */
+  public static oracleSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.ORACLE_SINGLE_USER,
+    });
+  }
+
+  /**
+   * Oracle Multi User
+   */
+  public static oracleMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.ORACLE_MULTI_USER,
+    });
+  }
+
+  /**
+   * MariaDB Single User
+   */
+  public static mariaDbSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MARIADB_SINGLE_USER,
+    });
+  }
+
+  /**
+   * MariaDB Multi User
+   */
+  public static mariaDbMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MARIADB_MULTI_USER,
+    });
+  }
+
+  /**
+   * SQL Server Single User
+   */
+  public static sqlServerSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.SQLSERVER_SINGLE_USER,
+    });
+  }
+
+  /**
+   * SQL Server Multi User
+   */
+  public static sqlServerMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.SQLSERVER_MULTI_USER,
+    });
+  }
+
+  /**
+   * Redshift Single User
+   */
+  public static redshiftSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.REDSHIFT_SINGLE_USER,
+    });
+  }
+
+  /**
+   * Redshift Multi User
+   */
+  public static redshiftMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.REDSHIFT_MULTI_USER,
+    });
+  }
+
+  /**
+   * MongoDB Single User
+   */
+  public static mongoDbSingleUser(options: SingleUserHostedRotationOptions = {}) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MONGODB_SINGLE_USER,
+    });
+  }
+
+  /**
+   * MongoDB Multi User
+   */
+  public static mongoDbMultiUser(options: MultiUserHostedRotationOptions) {
+    return new HostedRotation({
+      ...options,
+      type: HostedRotationType.MONGODB_MULTI_USER,
+    });
+  }
+
+  private securityGroups?: ec2.ISecurityGroup[];
+
+  constructor(private readonly props: HostedRotationProps) {
```

Comment:
```suggestion
  private constructor(private readonly props: HostedRotationProps) {
```

If we want to offer future extension, we can do it via `customSingleUser` / `customMultiUser` (or just a single `custom`) factory method. That being said, I don't see this as one of the enum-like areas where there is going to be a lot of extension; new database engines don't come out all that often.